Src/Particle/AMReX_ParticleContainerI.H

```diff
@@ -772,6 +772,61 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
     }
 }
 
+struct AssignGridFilter
+{
+
+    int m_lev_min, m_lev_max, m_nGrow, m_gid;
+    amrex::AmrAssignGrid<amrex::DenseBinIteratorFactory<amrex::Box>> m_assign_grid;
+
+    /**
+       \brief This filters based on matching grids
+       *
+     **/
+    AssignGridFilter (amrex::AmrAssignGrid<amrex::DenseBinIteratorFactory<amrex::Box>> assign_grid, int gid, int level, int nGrow)
+      : m_lev_min(level), m_lev_max(level+1), m_nGrow(nGrow), m_gid(gid), m_assign_grid(assign_grid)
+    {}
+
+    template <typename SrcData>
+    AMREX_GPU_HOST_DEVICE
+    int operator() (const SrcData& src, int src_i) const noexcept
+    {
+        const auto tup_min = (m_assign_grid)(src.m_aos[src_i], m_lev_min, m_lev_max, m_nGrow);
+        const auto tup_max = (m_assign_grid)(src.m_aos[src_i], m_lev_max, m_lev_max, m_nGrow);
+        const auto p_boxes = amrex::get<0>(tup_min);
+        const auto p_boxes_max = amrex::get<0>(tup_max);
+        const auto p_levs_max  = amrex::get<1>(tup_max);
+        return p_boxes_max >=0 && p_boxes == m_gid && p_levs_max == m_lev_max;
+    }
+};
+
+struct TransformerGhost
+{
+
+    /**
+       \brief This copies the particle but multiplies all the idata by m_factor
+       *
+     **/
+    TransformerGhost ()
+    {}
+
+    template <typename DstData, typename SrcData>
+    AMREX_GPU_HOST_DEVICE
+    void operator() (DstData& dst, const SrcData& src,
+                     int src_i, int dst_i) const noexcept
+    {
+        dst.m_aos[dst_i] = src.m_aos[src_i];
+        for (int j = 0; j < DstData::NAR; ++j)
+            dst.m_rdata[j][dst_i] = src.m_rdata[j][src_i];
+        for (int j = 0; j < dst.m_num_runtime_real; ++j)
+            dst.m_runtime_rdata[j][dst_i] = src.m_runtime_rdata[j][src_i];
+        for (int j = 0; j < DstData::NAI; ++j)
+            dst.m_idata[j][dst_i] = src.m_idata[j][src_i];
+        for (int j = 0; j < dst.m_num_runtime_int; ++j)
+            dst.m_runtime_idata[j][dst_i] = src.m_runtime_idata[j][src_i];
+        ParticleIDWrapper((dst.m_aos[dst_i]).m_idcpu) = GhostParticleID;
```

Comment:
I think you can do `dst.m_aos[dst_i].id() = GhostParticleID;`

Reply:
Pushed update commit, using id directly resulted in strange cpu or id values when debugging at one point, but I can't reproduce that behavior anymore